state-chain/runtime/src/chainflip.rs

```diff
@@ -249,28 +244,20 @@ macro_rules! impl_transaction_builder_for_evm_chain {
 				)
 			}
 
-			/// Calculate the gas limit for a this evm chain's call, using the current gas price.
-			/// Currently for only CCM calls, the gas limit is calculated as:
-			/// Gas limit = gas_budget / (multiplier * base_gas_price + priority_fee)
-			/// All other calls uses a default gas limit. Multiplier=1 to avoid user overpaying for gas.
-			/// The max_fee_per_gas will still have the default this evm chain's base fee multiplier applied.
+			/// Calculate the gas limit for this evm chain's call. This is only for CCM calls.
 			fn calculate_gas_limit(call: &$chain_api<$env>) -> Option<U256> {
-				if let Some(gas_budget) = call.gas_budget() {
-					let current_fee_per_gas = $chain_tracking::chain_state()
-						.or_else(||{
-							log::warn!("No chain data for {}. This should never happen. Please check Chain Tracking data.", $chain::NAME);
-							None
-						})?
-						.tracked_data
-						.max_fee_per_gas(One::one());
-					Some(gas_budget
-						.checked_div(current_fee_per_gas)
-						.unwrap_or_else(||{
-							log::warn!("Current gas price for {} is 0. This should never happen. Please check Chain Tracking data.", $chain::NAME);
-							Default::default()
-						}).min($max_gas_limit)
-						.into())
+				if let (Some(gas_budget), Some(message_length), Some(transfer_asset), Some(native_asset)) = (call.gas_budget(), call.message_length(), call.transfer_asset(), <$env as EvmEnvironmentProvider<$chain>>::token_address($chain::GAS_ASSET)) {
```

Comment:
> is there any reason these all have to be accessed separately?

Just fixed that.

> calculate_ccm_gas_limit could also take an asset instead of bool

The issue there is that from the `call.transfer_asset()` we just get H160 (address of the token) but not the `Asset`, given that the call has the encoded data already for EVM.  Comparing them here, where we can get the token addresses from the `EvmEnvironmentProvider` is quite easy and then pass a bool. Passing the token address to that `calculate_ccm_gas_limit` is quite ugly too. Is there an existing way to get the asset from a token address so we can compare them here and pass the asset to `calculate_gas_limit`?
Otherwise I guess I can just add a function to the `EvmEnvironmentProvider` like `token_address_to_asset`.

Reply:
Ok yeah i didn't quite realise we were dealing with addresses here.